pkg/storageincentives/agent.go

```diff
@@ -302,6 +305,41 @@ func (a *Agent) start(blockTime time.Duration, blocksPerRound, blocksPerPhase ui
 	}
 }
 
+func purgeStaleDataHandler(logger log.Logger, store storage.StateStorer, currentRound uint64) {
+	if currentRound <= purgeStaleDataThreshold {
+		return
+	}
+
+	purgeRound := func(round uint64) error {
+		return errors.Join(
+			removeCommitKey(store, round),
+			removeRevealRound(store, round),
+			removeSample(store, round),
+		)
+	}
+
+	from, err := getLastPurgedRound(store)
+	if err != nil {
+		logger.Error(err, "failed getting last purged round")
+		return
+	}
+
+	to := currentRound - purgeStaleDataThreshold
+
+	for i := from; i < to; i++ {
```

Comment:
this `from` is problematic because it will start from zero and iterate until a large number

What I recommend is to remove the `getLastPurgedRound` and `saveLastPurgedRound` and tackle the problem this way:

for each commit, reveal, and sample entries, store the round as well, and make the keys sortable by round number (this may be the case already because you append the round number to the key)
and when it's time to purge, iterate on the each prefix (commit, reveal, and sample) and delete the entry if the round is within the threshold.

Reply:
agree, that's more neater approach :+1: 

PR was updated, please review again